### PR TITLE
Jsonifying heat emission, and hot springs

### PR DESCRIPTION
**Pull request by Mom-Bun**
#### Summary

SUMMARY: Content "Hot Springs"

#### Purpose of change

Currently the code that governs the heat emission that lava gives off is hardcoded and restricted to "tr_lava" This code jsonifies it into an optional modifier like move_cost, and since the New England area has hot springs, I've added that as an example of it.

#### Describe the solution

I had some help configuring with the code (Thanks Candlebury!) but I basically made an optional terrain modifier in the same vein as move_cost, and editing the heat code in game.cpp to move it away from applying to "tr_lava" to "heat_radiation"

I also edited lava to "heat_radiation": 3 to represent how they were before, and added hot spring water, and hot springs. Currently the hot springs are only 1 intensity, and don't have much use, and can still burn you, so they aren't a nice steaming bath.

#### Describe alternatives you've considered

I've debated forgoing the hot springs part of this, and just making this PR about jsonifying heat emissions, but I figured it would be a good idea to include an example of the usage of it besides the current in game lava.

#### Testing

Compiles successfully, and the hot springs spawn similar (albeit rarer) than normal ponds, and both lava and the hot springs emit heat and can still hurt you.

### CODE REVIEW

File: data/json/furniture_and_terrain/terrain-liquids.json [hunk 1]
```diff
@@ -391,5 +392,18 @@
     },
     "connects_to": "WATER",
     "examine_action": "water_source"
+  },
+  {
+    "type": "terrain",
+    "id": "t_water_hot",
+    "name": "hot spring water",
+    "description": "Scalding hot water, not particulary safe for swimming. With a watertight container, you could gather fresh water from here.  Not safe to drink as is.",
```
Comment (Pupsi-Mupsi) [inline data/json/furniture_and_terrain/terrain-liquids.json:400]:
```suggestion
    "description": "Scalding hot water, not particulary safe for swimming.  With a watertight container, you could gather fresh water from here.  Not safe to drink as is.",
```

File: data/json/overmap/overmap_special/specials.json [hunk 2]
```diff
@@ -3558,6 +3558,15 @@
     "occurrences": [ 0, 30 ],
     "flags": [ "CLASSIC", "WILDERNESS" ]
   },
+  {
+    "type": "overmap_special",
+    "id": "Hot Springs",
+    "overmaps": [ { "point": [ 0, 0, 0 ], "overmap": "hot_springs_north" } ],
+    "locations": [ "field" ],
+    "city_distance": [ 1, -1 ],
+    "occurrences": [ 0, 30 ],
```
Comment (Night-Pryanik) [inline data/json/overmap/overmap_special/specials.json:3567]: Are hot springs that much rife in New England? Up to 30 locations per overmap feels like way too much for them.

Comment (kevingranade) [reply data/json/overmap/overmap_special/specials.json:3567]: Yea they seem fairly rare in the region.

Comment (Positronic-Girl) [reply data/json/overmap/overmap_special/specials.json:3567]: Oh, this was supposed to be rarer, yeah that is way too much!
What would be a better occurance rate?

Comment (Night-Pryanik) [reply data/json/overmap/overmap_special/specials.json:3567]: I'm not sure, but something like up to 1 or 2 per overmap.

Comment (Positronic-Girl) [reply data/json/overmap/overmap_special/specials.json]: Set it at 2, which should possibly make it rarer

**Pull request merged**